File: resource-registry/src/test/java/nl/haploid/octowight/registry/TestData.java
package nl.haploid.octowight.registry;

import nl.haploid.octowight.AtomChangeEvent;
import nl.haploid.octowight.AtomGroup;
import nl.haploid.octowight.registry.data.ResourceRoot;
import nl.haploid.octowight.registry.repository.ResourceElementDmo;
import nl.haploid.octowight.registry.repository.ResourceModelDmo;
import nl.haploid.octowight.registry.repository.ResourceModelId;
import nl.haploid.octowight.registry.repository.ResourceRootDmo;

import java.util.Random;
import java.util.UUID;

public class TestData {

	public static long nextLong() {
		return new Random().nextLong();
	}

	public static String nextString() {
		return UUID.randomUUID().toString();
	}

	public static ResourceRoot resourceRoot() {
		return resourceRoot(nextLong());
	}

	public static ResourceRoot resourceRoot(final Long resourceId) {
		final ResourceRoot resourceRoot = new ResourceRoot();
		resourceRoot.setResourceId(resourceId);
		resourceRoot.setResourceType("olson");
		resourceRoot.setAtomId(nextLong());
		resourceRoot.setAtomOrigin("madison avenue");
		resourceRoot.setAtomType("draper");
		return resourceRoot;
	}

	public static AtomChangeEvent atomChangeEvent(final AtomGroup atomGroup) {
		final AtomChangeEvent event = new AtomChangeEvent();
		event.setId(nextLong());
		event.setAtomId(nextLong());
		event.setAtomOrigin(atomGroup.getAtomOrigin());
		event.setAtomType(atomGroup.getAtomType());
		return event;
	}

	public static AtomChangeEvent atomChangeEvent(final String atomType) {
		final AtomChangeEvent event = new AtomChangeEvent();
		event.setId(nextLong());
		event.setAtomId(nextLong());
		event.setAtomOrigin("everywhere");
		event.setAtomType(atomType);
		return event;
	}

	public static ResourceRootDmo resourceRootDmo() {
		return resourceRootDmo("olson");
	}

	public static ResourceRootDmo resourceRootDmo(final String resourceType) {
		final ResourceRootDmo dmo = new ResourceRootDmo();
		dmo.setResourceId(nextLong());
		dmo.setResourceType(resourceType);
		dmo.setAtomId(nextLong());
		dmo.setAtomOrigin("madison avenue");
		dmo.setAtomType("draper");
		dmo.setVersion(nextLong());
		return dmo;
	}

	public static ResourceElementDmo resourceElementDmo(final ResourceRootDmo resourceRootDmo, final AtomChangeEvent atomChangeEvent) {
		final ResourceElementDmo dmo = new ResourceElementDmo();
		dmo.setResourceId(resourceRootDmo.getResourceId());
		dmo.setResourceType(resourceRootDmo.getResourceType());
		dmo.setAtomId(atomChangeEvent.getAtomId());
		dmo.setAtomOrigin(atomChangeEvent.getAtomOrigin());
		dmo.setAtomType(atomChangeEvent.getAtomType());
		return dmo;
	}

	public static ResourceElementDmo resourceElementDmo() {
		final ResourceElementDmo dmo = new ResourceElementDmo();
		dmo.setResourceId(nextLong());
		dmo.setResourceType("creative director");
		dmo.setAtomId(nextLong());
		dmo.setAtomOrigin("madison avenue");
		dmo.setAtomType("draper");
		return dmo;
	}

	public static ResourceModelDmo resourceModelDmo() {
		final ResourceModelDmo expectedResourceModelDmo = new ResourceModelDmo();
		final ResourceModelId resourceModelId = new ResourceModelId();
		resourceModelId.setResourceId(TestData.nextLong());
		resourceModelId.setResourceType(TestData.nextString());
		expectedResourceModelDmo.setId(resourceModelId);
		expectedResourceModelDmo.setBody(TestData.nextString());
		return expectedResourceModelDmo;
	}

	public static AtomGroup atomGroup() {
		final AtomGroup atomGroup = new AtomGroup();
		atomGroup.setAtomOrigin("new york");
		atomGroup.setAtomType("advertising agency");
		return atomGroup;
	}

	public static ResourceModelId resourceModelId() {
		final ResourceModelId resourceModelId = new ResourceModelId();
		resourceModelId.setResourceId(nextLong());
		resourceModelId.setResourceType(nextString());
		return resourceModelId;
	}
}

File: resource-registry/src/test/scala/nl/haploid/octowight/registry/TestData.scala
package nl.haploid.octowight.registry

import java.util.{Random, UUID}

import nl.haploid.octowight.registry.data.{Atom, ResourceIdentifier, ResourceRoot}
import nl.haploid.octowight.registry.repository._
import nl.haploid.octowight.{AtomChangeEvent, AtomGroup}

object TestData {

  def atom = new Atom(nextLong, nextString, nextString)

  def atomChangeEvent(atomCategory: String) =
    AtomChangeEvent(id = nextLong, origin = nextString, category = atomCategory)

  def atomChangeEvent(atomGroup: AtomGroup) =
    AtomChangeEvent(id = nextLong, origin = atomGroup.origin, category = atomGroup.category)

  def atomGroup = new AtomGroup(origin = TestData.nextString, category = TestData.nextString)

  def nextLong: Long = new Random().nextLong

  def nextString = UUID.randomUUID.toString

  // TODO: not random, so this could be an apply method
  def resourceElementDmo(resourceRootDmo: ResourceRootDmo, atomChangeEvent: AtomChangeEvent) = {
    val atomDmo: AtomDmo = AtomDmo(new Atom(atomChangeEvent.id, atomChangeEvent.origin, atomChangeEvent.category))
    new ResourceElementDmo(resourceId = resourceRootDmo.resourceId, resourceCollection = resourceRootDmo.resourceCollection, atom = atomDmo)
  }

  def resourceElementDmo =
    new ResourceElementDmo(resourceId = nextLong, resourceCollection = nextString, atom = AtomDmo(atom))

  def resourceIdentifier = new ResourceIdentifier(collection = nextString, id = nextLong)

  def resourceModelDmo = new ResourceModelDmo(id = resourceModelDmoId, body = nextString, version = nextLong)

  def resourceModelDmoId = new ResourceModelDmoId(resourceId = nextLong, resourceCollection = nextString)

  def resourceRoot: ResourceRoot = resourceRoot(nextLong)

  def resourceRoot(resourceId: Long) =
    new ResourceRoot(resourceId = resourceId, resourceCollection = nextString, root = atom, version = null)

  def resourceRootDmo: ResourceRootDmo = resourceRootDmo(nextString)

  def resourceRootDmo(resourceCollection: String) =
    new ResourceRootDmo(
      resourceId = nextLong,
      resourceCollection = resourceCollection,
      root = AtomDmo(atom),
      version = nextLong)
}
